python/lsst/analysis/tools/tasks/propertyMapTractAnalysis.py

```diff
@@ -0,0 +1,179 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = [
+    "PropertyMapTractAnalysisConfig",
+    "PropertyMapTractAnalysisTask",
+]
+
+from typing import Mapping
+
+from lsst.daf.butler import DataCoordinate
+from lsst.pex.config import Config, ConfigDictField, Field, ListField
+from lsst.pipe.base import ButlerQuantumContext, InputQuantizedConnection, OutputQuantizedConnection
+from lsst.pipe.base import connectionTypes as ct
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class PropertyMapConfig(Config):
+    coaddName = Field(
+        dtype=str,
+        doc="Coadd name: typically one of deep or goodSeeing.",
+        default="deep",
+    )
+    operations = ListField(
+        dtype=str,
+        doc="List of operations whose corresponding maps should be retrieved.",
+        default=["min", "max", "mean", "weighted_mean", "sum"],
+    )
+
+
+class PropertyMapTractAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("skymap", "tract", "band"),
+    defaultTemplates={"outputName": "propertyMapTract"},
+):
+    healSparsePropertyMapsConfig = ct.Input(
+        doc="Configuration parameters for HealSparseInputMapTask in pipe_tasks.",
+        name="healSparsePropertyMaps_config",
+        storageClass="Config",
+        dimensions=(),
+    )
+
+    skymap = ct.Input(
+        doc="The skymap that covers the tract that the data is from.",
+        name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+        storageClass="SkyMap",
+        dimensions=("skymap",),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        operationNameLookup = {
+            "min": "Minimum",
+            "max": "Maximum",
+            "mean": "Mean",
+            "weighted_mean": "Weighted mean",
+            "sum": "Sum",
+        }
+
+        # Making connections for the maps that are configured to run.
+        for propertyName in config.properties:
+            coaddName = config.properties[propertyName].coaddName
+            for operationName in config.properties[propertyName].operations:
+                operationLongName = operationNameLookup[operationName]
+                name = f"{coaddName}Coadd_{propertyName}_map_{operationName}"
+                setattr(
+                    self,
+                    f"{coaddName}Coadd_{propertyName}_{operationName}",
+                    ct.Input(
+                        doc=f"{operationLongName}-value map of {{propertyLongName}}",
+                        name=name,
+                        storageClass="HealSparseMap",
+                        dimensions=("tract", "skymap", "band"),
+                        multiple=False,
+                        deferLoad=False,
```

Comment:
For a small number of tract maps, it may not be a problem.  But I always do `deferLoad=True` to make sure we don't hold too much in memory at once because they all get their own plots.

Reply:
Well, you were the one who wrote the validation code, so that can be fixed.  But I suppose you would also want to check the type contained in the deferred dataset handle.  This can be done with `.storageClass`.  But at the same time I don't know if using deferred handles is some sort of analysis tools anti-pattern.

Reply:
Done!